Sources/Brave/Frontend/Browser/Playlist/Controllers/PlaylistListViewController+TableViewDelegate.swift

```diff
@@ -173,7 +174,15 @@ extension PlaylistListViewController: UITableViewDelegate {
           UIAlertAction(
             title: Strings.PlayList.sharePlaylistOpenInNewPrivateTabTitle, style: .default,
             handler: { [weak self] _ in
-              self?.openInNewTab(currentItem, isPrivate: true)
+              if !isPrivateBrowsing, Preferences.Privacy.privateBrowsingLock.value {
+                self?.askForLocalAuthentication { [weak self] success, error in
+                  if success {
+                    self?.openInNewTab(currentItem, isPrivate: true)
+                  }
+                }
+              } else {
+                self?.openInNewTab(currentItem, isPrivate: true)
+              }
```

Comment:
There's an additional action to open in a new private tab in playlist by table view context menu: https://github.com/brave/brave-ios/blob/15743fea2949291b5a167853c4b6b86d3e312643/Sources/Brave/Frontend/Browser/Playlist/Controllers/PlaylistListViewController%2BTableViewDelegate.swift#L270

You may want to unify these to call one shared method

Reply:
Fixed at https://github.com/brave/brave-ios/pull/7959/commits/68dbdac411e43999da1077d4dc34a50e45e5d03c